### PR TITLE
HeatmapTooltip: Refactor to simplify upcoming PRs

### PR DESCRIPTION
**Pull request by leeoniya**
plz test this with **without** the `newVizTooltips` feature flag.

reduces the diff of https://github.com/grafana/grafana/pull/75712

this pulls out changes that are common or will be shared between current and new tooltips, including changes demanded by betterer.

### CODE REVIEW

File: public/app/plugins/panel/heatmap/HeatmapHoverView.tsx [hunk 1]
```diff
@@ -287,3 +236,68 @@ const HeatmapHoverCell = ({ data, hover, showHistogram, scopedVars, replaceVars
     </>
   );
 };
+
+export function renderHistogram(
```
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/HeatmapHoverView.tsx:240]: this was all extracted for reuse with only non-effectual changes, except the new bar colors (below)

File: public/app/plugins/panel/heatmap/HeatmapHoverView.tsx [hunk 2]
```diff
@@ -287,3 +236,68 @@ const HeatmapHoverCell = ({ data, hover, showHistogram, scopedVars, replaceVars
...
+      j++;
+    }
+
+    histCtx.clearRect(0, 0, histCanWidth, histCanHeight);
+
+    histCtx.fillStyle = '#2E3036';
+    histCtx.fill(pRest);
+
+    histCtx.fillStyle = '#5794F2';
+    histCtx.fill(pHov);
```
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/HeatmapHoverView.tsx:301]: the hex colors here were updated with new tooltip style

File: public/app/core/components/ColorScale/ColorScale.tsx
Comment (leeoniya) [inline public/app/core/components/ColorScale/ColorScale.tsx:1]: one-to-one refactor to using style objects

File: public/app/plugins/panel/heatmap/HeatmapHoverView.tsx (hunk 2 above)
Comment (leeoniya) [reply public/app/plugins/panel/heatmap/HeatmapHoverView.tsx]: (moved to new file)

**Review by adela-almasan — APPROVED** [review]

**Pull request merged**